Automattic/wp-calypso
# CPT: Display warning when post type not supported for site

This pull request seeks to accurately reflect when a post type is not supported for a site. Further, it directs users to the site settings page for managing types which can be enabled for a site, with a fallback for post types that are unknown (such as when switching sites while viewing a listing of custom post types for a Jetpack site).

Before|After
---|---
![Before](https://cloud.githubusercontent.com/assets/1779930/15162819/ffed3a20-16d2-11e6-8a2d-8ec879a5b7c0.png)|![After](https://cloud.githubusercontent.com/assets/1779930/15162826/0d1b6f1e-16d3-11e6-9f28-4cb7c085ec34.png)

__Testing instructions:__

Verify that usage is unaffected for post types supported by a site. Verify that post types which can be enabled (portfolio or testimonial) prompt managing site settings. Otherwise, note a "Content type unsupported" fallback message is shown.

(These issues become particularly noticeable when switching between sites while managing custom post types)

1. Navigate to My Sites
2. If the current site supports a custom post type, click the sidebar navigation for that type
3. Note that the posts listing displays as it does in the master branch
4. Switch to a site where the custom post type is not supported
5. Note that...
 - If the post type is "portfolio" or "testimonials", you are prompted to manage your site settings
 - Otherwise, "Content type unsupported" warning is shown

__Caveats:__

- The writing settings page does not yet include settings for managing custom post types (#4502)

__Follow-up tasks:__

- The screen title should be improved for these pages. It currently shows an empty title with site suffix.
